pallets/sponsorship/src/lib.rs

```diff
@@ -278,7 +279,7 @@ pub mod pallet {
 			reserve_quota: BalanceOf<T>,
 		) -> DispatchResult {
 			let who = ensure_signed(origin)?;
-			ensure!(!Self::migration_in_progress(), Error::<T>::MigrationInProgress);
+			Self::ensure_migration_completed()?;
```

Comment:
We don't need to migration logic anymore, do we? Suggest you leave taking out the migration logic to another PR.